src/main/groovy/io/sentry/android/gradle/GenerateSentryProguardUuidTask.java

```diff
@@ -0,0 +1,54 @@
+package io.sentry.android.gradle;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.UUID;
+
+abstract class GenerateSentryProguardUuidTask extends DefaultTask {
+    @OutputDirectory
+    public abstract DirectoryProperty getOutputDirectory();
+
+    @Internal
+    public Provider<UUID> getOutputUuid() {
+        return getOutputUuidInternal();
+    }
+
+    @Internal
+    protected abstract Property<UUID> getOutputUuidInternal();
+
+    @Internal
+    private final Provider<RegularFile> outputFile = getOutputDirectory().map(dir -> dir.file("sentry-debug-meta.properties"));
+
+    public GenerateSentryProguardUuidTask() {
+        getOutputs().upToDateWhen(spec -> false);
+        setDescription("Generates a unique build ID");
+    }
+
+    @Internal
+    public Provider<RegularFile> getOutputFile() {
+        return outputFile;
+    }
+
+    @TaskAction
+    protected void generateProperties() throws IOException {
+        UUID uuid = UUID.randomUUID();
+        getOutputUuidInternal().set(uuid);
+
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(outputFile.get().getAsFile())))) {
+            writer.write("io.sentry.ProguardUuids=");
+            writer.write(uuid.toString());
+        }
```

Comment:
I'm wondering if this would cause some problems like:

my mapping file has changed (not the same hash), but we didn't generate a new UUID and we've overwritten the mapping file on the server (if the server allows doing so).

or just the other way around, we've generated a new UUID, but the mapping file is still the same, so today `sentry-cli` takes care of it by computing the hash of the file, but right now we'd be doing this here instead, although not checking the hash at all.

https://github.com/getsentry/sentry-cli/blob/762ff34376a380789b3de731764836d50c7e0106/src/commands/upload_proguard.rs#L117-L131

Reply:
The first issue has already been addressed by the `outputs.upToDateWhen { false }`. This means that the task will never be up to date and every build will generate a new UUID.

The second part I can't answer since it depends on how the CLI/backend behaves. If things break when uploading the same mappings with a different UUID we can work around that by copying the mappings file and appending a comment that includes the UUID to ensure that the hash would be different.

Reply:
However, reading the documentation for `--uuid` it does not say that you cannot use the same mappings for multiple builds.

Reply:
I'll double-check with the `sentry-cli` team, reading Rust is not that easy hehe

Reply:
@ansman this is currently blocked by https://github.com/getsentry/sentry-cli/issues/914
the situation is: we'll generate new UUIDs (for every build), if the mapping file has not changed (the hash itself), the UUID won't exist on our backend (because we won't upload it again, so the new UUID won't be valid) hence stack traces won't be demangled, we'll get somebody into it asap.